src/cryptonote_core/loki_name_system.cpp

```diff
@@ -0,0 +1,1124 @@
+            auto *name    = reinterpret_cast<char const *>(sqlite3_column_text(statement, static_cast<int>(mapping_record_row::name)));
+
+            size_t value_len = static_cast<size_t>(sqlite3_column_bytes(statement, static_cast<int>(mapping_record_row::value)));
+            auto *value      = reinterpret_cast<char const *>(sqlite3_column_text(statement, static_cast<int>(mapping_record_row::value)));
+
+            tmp_entry.name  = std::string(name, name_len);
+            tmp_entry.value = std::string(value, value_len);
+
+            if (!sql_copy_blob(statement, static_cast<int>(mapping_record_row::txid), tmp_entry.txid.data, sizeof(tmp_entry.txid)))
+              return false;
+
+            if (!sql_copy_blob(statement, static_cast<int>(mapping_record_row::prev_txid), tmp_entry.prev_txid.data, sizeof(tmp_entry.prev_txid)))
+              return false;
+
+            data_loaded = true;
+            if (type == lns_sql_type::get_mappings_by_user || type == lns_sql_type::get_mappings_on_height_and_newer)
+            {
+              auto *records = reinterpret_cast<std::vector<mapping_record> *>(context);
+              records->emplace_back(std::move(tmp_entry));
+            }
+            else
+            {
+              mapping_record *entry = reinterpret_cast<mapping_record *>(context);
+              *entry                = std::move(tmp_entry);
+            }
+          }
+          break;
+        }
+      }
+      break;
+
+      case SQLITE_BUSY: break;
+      case SQLITE_DONE:
+      {
+        infinite_loop = false;
+        result        = (type > lns_sql_type::get_sentinel_start && type < lns_sql_type::get_sentinel_end) ? data_loaded : true;
+        break;
+      }
+
+      default:
+      {
+        LOG_PRINT_L1("Failed to execute statement: " << sqlite3_sql(statement) <<", reason: " << sqlite3_errstr(step_result));
+        infinite_loop = false;
+        break;
+      }
+    }
+  }
+
+  sqlite3_reset(statement);
+  sqlite3_clear_bindings(statement);
+  return result;
+}
+
+bool mapping_record::active(cryptonote::network_type nettype, uint64_t blockchain_height) const
+{
+    if (!loaded) return false;
+    if (type != static_cast<uint16_t>(mapping_type::lokinet)) return true;
+    uint64_t expiry_blocks            = lns::lokinet_expiry_blocks(nettype);
+    uint64_t const last_active_height = register_height + expiry_blocks;
+    return last_active_height >= (blockchain_height - 1);
+}
+
+static bool sql_compile_statement(sqlite3 *db, char const *query, int query_len, sqlite3_stmt **statement)
+{
+  int prepare_result = sqlite3_prepare_v2(db, query, query_len, statement, nullptr);
+  bool result        = prepare_result == SQLITE_OK;
+  if (!result) MERROR("Can not compile SQL statement: " << query << ", reason: " << sqlite3_errstr(prepare_result));
+  return result;
+}
+
+uint64_t burn_requirement_in_atomic_loki(uint8_t hf_version)
+{
+  (void)hf_version;
+  return 30 * COIN;
+}
```

Comment:
This 30 value seems way too significant to be hard-coded here.  Rather we should have some named constants for it in cryptonote_config, but also split up by type (and then pass `type` into this function):

- LNS_BURN_LOKINET_1Y
- LNS_BURN_SESSION
- LNS_BURN_WALLET
- LNS_BURN_CUSTOM

with them all set to 30 for the purpose of this PR.  (I think we, i.e. Loki community more generally, should discuss further appropriate pricing for these; 30 seems okay for a .loki registration, but seems really expensive for a Session or wallet address registration).